src/app/login/login.component.ts

```diff
@@ -32,30 +32,36 @@ require('./login.scss');
         </div>
     `,
     styleUrls:['./login.scss']
-    
+
 })
-export class LoginComponent { 
+export class LoginComponent {
     constructor(
         private couchService: CouchService,
-        private router: Router
+        private router: Router,
+        private route: ActivatedRoute
     ) {}
-    
+
+    ngOnInit() {

```

Comment:
I don't think you actually should use ngOnInit here, just assign returnUrl to the route.

ngOnInit is used mainly for accessing the services to initialize data.  `returnUrl` can be set with just simple assignment, so it doesn't need to utilize that lifecycle hook.

Reply:
Used returnUrl in constructor